File: 0771-jewels-and-stones/0771-jewels-and-stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        std::unordered_map<char, int> stoneMap;
        for(int i = 0; i < stones.size(); i++) {
            char stone = stones[i];
            if(stoneMap.find(stone) != stoneMap.end()) {
                stoneMap[stone]++;
            } else {
                stoneMap[stone] = 1;
            }
        }
        int numberOfJewelStones = 0;
        for(int i = 0; i < jewels.size(); i++) {
            char jewel = jewels[i];
            numberOfJewelStones += stoneMap[jewel];
        }
        return numberOfJewelStones;
    }
};

File: 0771-jewels-and-stones/0771-jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        stone_dict = {}
        for stone in stones:
            if stone in stone_dict:
                stone_dict[stone] += 1
            else:
                stone_dict[stone] = 1
        number_of_jewel_stones = 0
        for jewel in jewels:
            if jewel in stone_dict:
                number_of_jewel_stones += stone_dict[jewel]
        return number_of_jewel_stones